src/zmqsubscriber/zmqsubscriber.go

```diff
@@ -80,17 +80,29 @@ func NewZMQSubscriber(host string, port string) (*ZMQSubscriber, error) {
 	}, nil
 }
 
+// Listen for new messages, parse messages into native data types and put them into channels
+// `IncomingTx` and `IncomingBlocks`.
+// Returns error if something goes wrong and `nil` if stopped using `Stop()`
 func (z *ZMQSubscriber) Run() error {
 	defer func () {
 		if err := z.socket.Close(); err != nil {
 			log.Printf("[ZMQ] socket closed with error %q (ignored)", err)
 		}
 	}()
 
+	parseErrors := make(chan error)
+
 	// FIXME(#11):  Workaround for bug where zmq crashes when Close() is called from a different
 	// 				goroutine. Instead of permanently blocking on Recv(), we set a timeout and check
 	// 				for `z.cancel`.
 	for !z.cancel {
+		// if a parse error happened, exit the loop
+		select {
+		case err := <-parseErrors:
+			return err
+		default:
+		}
```

Comment:
Wouldn't a alternative be to log the error but continue with normal operation? Could cause use to miss out on a lot of data? Or is `Run()` called again after an error?

Reply:
at this stage of development I would prefer a crash with high visibility instead of error logs which can be easy to miss